Production_scripts/B05a_Process_A&E_(AE2)_extract.R

```diff
@@ -0,0 +1,950 @@
+    year = year,
+    recid = "AE2"
+  ) %>%
+  ## Recode GP Practice into a 5 digit number ##
+  # assume that if it starts with a letter it's an English practice and so recode to 99995
+  convert_eng_gpprac_to_dummy(gpprac) %>%
+  # use the CHI postcode and if that is blank, then use the epi postcode.
+  mutate(postcode = if_else(!is.na(postcode_chi), postcode_chi, postcode_epi)) %>%
+  ## recode cypher HB codes ##
+  mutate(across(c(hbtreatcode, hbrescode), ~ case_when(
+    .x == "A" ~ "S08000015",
+    .x == "B" ~ "S08000016",
+    .x == "F" ~ "S08000029",
+    .x == "G" ~ "S08000021",
+    .x == "H" ~ "S08000022",
+    .x == "L" ~ "S08000023",
+    .x == "N" ~ "S08000020",
+    .x == "R" ~ "S08000025",
+    .x == "S" ~ "S08000024",
+    .x == "T" ~ "S08000030",
+    .x == "V" ~ "S08000019",
+    .x == "W" ~ "S08000028",
+    .x == "Y" ~ "S08000017",
+    .x == "Z" ~ "S08000026"
+  ))) %>%
+  ## Allocate the costs to the correct month ##
+  # Create month variable
+  mutate(month = strftime(record_keydate1, "%m")) %>%
+  # Allocated cost in correct month
+  mutate(
+    apr_cost = if_else(month == "04", cost_total_net, 0),
+    may_cost = if_else(month == "05", cost_total_net, 0),
+    jun_cost = if_else(month == "06", cost_total_net, 0),
+    jul_cost = if_else(month == "07", cost_total_net, 0),
+    aug_cost = if_else(month == "08", cost_total_net, 0),
+    sep_cost = if_else(month == "09", cost_total_net, 0),
+    oct_cost = if_else(month == "10", cost_total_net, 0),
+    nov_cost = if_else(month == "11", cost_total_net, 0),
+    dec_cost = if_else(month == "12", cost_total_net, 0),
+    jan_cost = if_else(month == "01", cost_total_net, 0),
+    feb_cost = if_else(month == "02", cost_total_net, 0),
+    mar_cost = if_else(month == "03", cost_total_net, 0)
+  ) %>%
+  # sort for linking on CUP marker
+  arrange(record_keydate1, keyTime1, case_ref_number)
+
+
+# Add value labels---------------------------------------------------
+
+# A&E value labels
+ae_clean <- ae_clean %>%
+  mutate(
+    ae_arrivalmode = factor(ae_arrivalmode,
+      levels = c("01", "02", "03", "04", "05", "06", "07", "08", "98", "99"),
+      labels = c(
+        "Ambulance (road) Excludes involvement of an A&E retrieval team.",
+        "Ambulance (air) Travel for all or any part of journey by an aircraft operating as an ambulance.",
+        "Ambulance and A&E retrieval team. Includes both road and air ambulance modes of transport.",
+        "Out of Hours transport OOH has arranged transport (includes PTS transport but not emergency ambulance).",
+        "Private transport Includes car, taxi, motorbike, bicycle, etc",
+        "Public transport Includes bus, train, etc.",
+        "Walking On foot.",
+        "Police/prison transport Patient is brought to A&E in a police or prison vehicle.",
+        "Other The true value to be recorded is not covered by any of the specific given categories. Includes mortuary van.",
+        "Not known"
+      )
+    ),
```

Comment:
Creating factors with a different label when the data is a character doesn't work / could break people's code. e.g.

```r
ae_clean %>% filter(ae_arrivalmode == "01")
```
Would usually work but won't after it's been turned into a factor. R drops the underlying data and replaces it with a number 1:n for the number of levels. 

We can discuss 😄

Reply:
@Moohan so remove the labels code? and keep numeric levels?